Dockerfile

```diff
@@ -0,0 +1,45 @@
+ARG BASE_IMAGE=ubuntu:22.04
+FROM ${BASE_IMAGE} AS build-stage
+
+ARG PJ_TAG=3.9.2
+
+###############################################################################
+# Install general dependencies.
+###############################################################################
+RUN apt update && apt install -y build-essential cmake git 
+
+###############################################################################
+# Compile PlotJuggler
+###############################################################################
+WORKDIR /plotjuggler_ws/src
+RUN git clone --branch ${PJ_TAG} https://github.com/facontidavide/PlotJuggler.git PlotJuggler
+# Install PlotJuggler dependencies.
+RUN apt update && apt -y install qtbase5-dev libqt5svg5-dev libqt5websockets5-dev \
```

Comment:
better merge this with  Line9 : that will prevent redownload each time there is a change on PJ tag and make the package install in one place.

Reply:
Even though it breaks notional separation, I do believe you're right: the merit of not having to redownload all these dependencies is great.